publisher/publisher.go

```diff
@@ -80,6 +81,14 @@ func (pub *publisher) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
 	r2.Description = "KairosDB port"
 	config.Add(r2)
 
+	r3, err := cpolicy.NewBoolRule("useDynamic", true)
```

Comment:
As you are not comfortable with go, I've prepared something. You can replace existing code of TestPublish with the code which I prepared, please see:
```golang
func (s *KairosSuite) TestPublish() {
	Convey("Given snap KairosDB publisher testing", s.T(), func() {
		metrics := []plugin.MetricType{
			plugin.MetricType{
				Namespace_: core.NewNamespace("foo"),
				Timestamp_: time.Now(),
				Tags_:      map[string]string{"hostname": "127.0.0.1"},
				Data_:      43,
			},
			plugin.MetricType{
				Namespace_: core.NewNamespace("bar").AddDynamicElement("dynamic_element", "dynamic element of namespace"),
				Timestamp_: time.Now(),
				Tags_:      map[string]string{"hostname": "127.0.0.1"},
				Data_:      44,
			},
		}

		Convey("Publishing dynamic elements of namespace as tags for metric", func() {
			var buf bytes.Buffer
			config := make(map[string]ctypes.ConfigValue)
			u, _ := url.Parse(s.server.URL)
			hostPort := strings.Split(u.Host, ":")
			config["host"] = ctypes.ConfigValueStr{Value: hostPort[0]}
			port, _ := strconv.Atoi(hostPort[1])
			config["port"] = ctypes.ConfigValueInt{Value: port}
			config["useDynamic"] = ctypes.ConfigValueBool{Value: true}

			kp := New()
			cp, _ := kp.GetConfigPolicy()
			cfg, _ := cp.Get([]string{""}).Process(config)

			Convey("Publish provided metrics", func() {

				buf.Reset()
				enc := gob.NewEncoder(&buf)
				enc.Encode(metrics)
				err := kp.Publish(plugin.SnapGOBContentType, buf.Bytes(), *cfg)
				So(err, ShouldBeNil)
			})
		})

		Convey("Publishing without dynamic elements of namespace as tags for metric", func() {
			var buf bytes.Buffer
			config := make(map[string]ctypes.ConfigValue)
			u, _ := url.Parse(s.server.URL)
			hostPort := strings.Split(u.Host, ":")
			config["host"] = ctypes.ConfigValueStr{Value: hostPort[0]}
			port, _ := strconv.Atoi(hostPort[1])
			config["port"] = ctypes.ConfigValueInt{Value: port}
			config["useDynamic"] = ctypes.ConfigValueBool{Value: false}

			kp := New()
			cp, _ := kp.GetConfigPolicy()
			cfg, _ := cp.Get([]string{""}).Process(config)

			Convey("Publish provided metrics", func() {

				buf.Reset()
				enc := gob.NewEncoder(&buf)
				enc.Encode(metrics)
				err := kp.Publish(plugin.SnapGOBContentType, buf.Bytes(), *cfg)
				So(err, ShouldBeNil)
			})
		})
	})
}
```
You can replace existing code of `TestPublish` with the code which I prepared.